libnetwork/networkdb/networkdb_test.go

```diff
@@ -155,21 +156,18 @@ func testWatch(t *testing.T, ch chan events.Event, ev interface{}, tname, nid, k
 	select {
 	case rcvdEv := <-ch:
 		assert.Check(t, is.Equal(fmt.Sprintf("%T", rcvdEv), fmt.Sprintf("%T", ev)))
-		switch typ := rcvdEv.(type) {
-		case CreateEvent:
-			assert.Check(t, is.Equal(tname, typ.Table))
-			assert.Check(t, is.Equal(nid, typ.NetworkID))
-			assert.Check(t, is.Equal(key, typ.Key))
-			assert.Check(t, is.Equal(value, string(typ.Value)))
-		case UpdateEvent:
-			assert.Check(t, is.Equal(tname, typ.Table))
-			assert.Check(t, is.Equal(nid, typ.NetworkID))
-			assert.Check(t, is.Equal(key, typ.Key))
-			assert.Check(t, is.Equal(value, string(typ.Value)))
-		case DeleteEvent:
-			assert.Check(t, is.Equal(tname, typ.Table))
-			assert.Check(t, is.Equal(nid, typ.NetworkID))
-			assert.Check(t, is.Equal(key, typ.Key))
+		if event, ok := rcvdEv.(Event); ok {
+			switch event.Type {
+			case driverapi.Create, driverapi.Update:
+				assert.Check(t, is.Equal(tname, event.Table))
+				assert.Check(t, is.Equal(nid, event.NetworkID))
+				assert.Check(t, is.Equal(key, event.Key))
+				assert.Check(t, is.Equal(value, string(event.Value)))
+			case driverapi.Delete:
+				assert.Check(t, is.Equal(tname, event.Table))
+				assert.Check(t, is.Equal(nid, event.NetworkID))
+				assert.Check(t, is.Equal(key, event.Key))
+			}
```

Comment:
Separate case for `Delete` was still needed here, because the `Delete` would keep the `Value` of the event from what it was before (which would be `test_updated_value` in this case); I didn't want that tone part of the test (i.e., on `Delete` the content of `Value` doesn't matter, so isn't / shouldn't be checked).